WalletWasabi.Gui/Controls/WalletExplorer/SendControlViewModel.cs

```diff
@@ -871,63 +872,64 @@ private bool TryParseUserFee(out decimal userFee)
 				&& userFee > 0;
 		}
 
-		public ErrorDescriptors ValidateUserFeeText()
+		private void ValidateUserFeeText(IValidationErrors errors)
 		{
-			return TryParseUserFee(out _)
-				? ErrorDescriptors.Empty
-				: new ErrorDescriptors(new ErrorDescriptor(ErrorSeverity.Error, "Invalid fee."));
+			if (!TryParseUserFee(out _))
+			{
+				errors.Add(ErrorSeverity.Error, "Invalid fee.");
+			}
 		}
 
-		public ErrorDescriptors ValidatePassword() => PasswordHelper.ValidatePassword(Password);
+		private void ValidatePassword(IValidationErrors errors) => PasswordHelper.ValidatePassword(errors, Password);
 
-		public ErrorDescriptors ValidateActiveAddress()
+		private void ValidateAddress(IValidationErrors errors)
 		{
 			if (string.IsNullOrWhiteSpace(Address))
 			{
-				return ErrorDescriptors.Empty;
+				return;
 			}
 
 			if (Address == CustomChangeAddress)
 			{
-				return new ErrorDescriptors(new ErrorDescriptor(ErrorSeverity.Error, "The active address and the change address cannot be the same."));
+				errors.Add(ErrorSeverity.Error, "The active address and the change address cannot be the same.");
 			}
 
 			if (AddressStringParser.TryParseBitcoinAddress(Address, Global.Network, out _))
 			{
-				return ErrorDescriptors.Empty;
+				return;
 			}
 
 			if (AddressStringParser.TryParseBitcoinUrl(Address, Global.Network, out _))
 			{
-				return ErrorDescriptors.Empty;
+				return;
 			}
 
-			return new ErrorDescriptors(new ErrorDescriptor(ErrorSeverity.Error, "Invalid address."));
+			errors.Add(ErrorSeverity.Error, "Invalid address.");
 		}
 
-		public ErrorDescriptors ValidateCustomChangeAddress()
+		private void ValidateCustomChangeAddress(IValidationErrors errors)
 		{
 			if (string.IsNullOrWhiteSpace(CustomChangeAddress))
 			{
-				return ErrorDescriptors.Empty;
+				return;
 			}
 
 			if (IsMax)
 			{
-				return new ErrorDescriptors(new ErrorDescriptor(ErrorSeverity.Error, "Spending whole coins does not generate change."));
+				errors.Add(ErrorSeverity.Error, "Spending whole coins does not generate change.");
 			}
 
 			if (Address == CustomChangeAddress)
 			{
-				return new ErrorDescriptors(new ErrorDescriptor(ErrorSeverity.Error, "The active address and the change address cannot be the same."));
+				errors.Add(ErrorSeverity.Error, "The active address and the change address cannot be the same.");
```

Comment:
Should be a return here after the add.

Reply:
Address can be invalid and at the same time it can be the same as the customchangeaddress?

Why do we support multiple validation errors if we never use them.